R/backend-postgres.R

```diff
@@ -96,7 +123,24 @@ sql_translate_env.PostgreSQLConnection <- function(con) {
           stop("Unrecognized arguments to `wday`", call. = FALSE)
         }
       },
-      yday = function(x) sql_expr(EXTRACT(DOY %FROM% !!x))
+      yday = function(x) sql_expr(EXTRACT(DOY %FROM% !!x)),
+      floor_date = function(x, unit = "seconds", week_start = NULL) {
```

Comment:
This should use `unit = c("seconds", "minute", ...)` and then use `match.arg()` in the body of the function; that will take care of the error checking too.